### ISSUE TITLE
Fetch types are incompatible with TypeScript

### ISSUE THREAD

**Issue by kettanaito** [nodejs/undici#1943]
## Bug Description

The `Request` interface exposed by Undici does not satisfy the same type from the global type definitions (`lib/dom`), and also missing properties according to the Fetch API specification:

https://github.com/nodejs/undici/blob/d8d9a96b6b8f856d9e01a1edfe22c3d83c6996ec/types/fetch.d.ts#L138

## Reproducible By

As an example, consider the following code: 

```ts
import { Request } from 'undicit'

function transform(req: globalThis.Request) {}

transform(new Request('/hello'))
// ^^^ Property 'referrer' is missing in type 'import(".../node_modules/undici/types/fetch").Request' but required in type 'Request'.ts(2345)
```

The `referrer` property is not present in the Undici's `Request` type definition while it's both required in `lib/dom` definitions and the [Fetch API specification](https://fetch.spec.whatwg.org/#request-class).

```ts
readonly attribute [USVString](https://webidl.spec.whatwg.org/#idl-USVString) [referrer](https://fetch.spec.whatwg.org/#dom-request-referrer);
```

## Expected Behavior

Fetch API primitives such as Request, Response and Headers, are compatible with the global 

## Logs & Screenshots

No applicable logs.

## Environment

- Node: v16.16.0
- Undici: ^5.20.0
- TypeScript: ^4.9.4

### Additional context

I understand that I'm setting `lib/dom` as the source of truth here but I expect that shouldn't matter since:

1. Both Undici and `lib/node` aim to implement the Fetch API specification.
2. Undici is objectively missing some properties in its type definitions defined in the spec, like `referrer`. 

I'd like to open a pull request to ensure this compliance after discussing this with the maintainers of this lib.

**Comment by KhafraDev** [comment]
Would you like to send in a PR to add the missing propert(y/ies) to Request?

**Comment by kettanaito** [comment]
@KhafraDev, absolutely! Will open a pull request tomorrow, we can follow up there. Thanks for a quick response on this.

**Comment by kettanaito** [comment]
Hey, @KhafraDev. I've opened a draft to improve typings of Undici. Would appreciate an early review of this and any potential discussion. Thanks.

**Comment by sharifzadesina** [comment]
It is interesting that this bug is not yet fixed.

**Comment by mcollina** [comment]
> It is interesting that this bug is not yet fixed. Can we do any help?

PRs are welcome!

**Comment by NatoBoram** [comment]
> PRs are welcome!

Isn't https://github.com/nodejs/undici/pull/1964 already open?

**Comment by mcollina** [comment]
Actually it's abandoned with failing tests. A fresh one would be better.
